fbgemm_gpu/bench/bench_utils.py

```diff
@@ -159,13 +159,28 @@ def benchmark_requests(
     # Can be used to clear model's stats after warmup for example.
     callback_after_warmup: Optional[Callable[[], None]] = None,
     periodic_logs: bool = False,
+    warmup_ms: Optional[int] = None,
 ) -> float:
     times = []
 
     # Run at least one warmup iteration to avoid the long cudaLaunchKernel time
     # for the first kernel
     num_warmups = num_warmups + 1 if num_warmups >= 0 else 1
 
+    # warm-up the GPU before profiling  
+    if warmup_ms:
+        indices, offsets, weights = requests[0].unpack_3()
+        start_events = torch.cuda.Event(enable_timing=True)
+        end_events = torch.cuda.Event(enable_timing=True)
+        elapsed_time_ms=0
+        while elapsed_time_ms < warmup_ms:
+            if torch.cuda.is_available():
```

Comment:
This "if" clause is unsafe, two ways to change it:
1. move it above "while"
2. add "else" and consider the CPU case.